README.md

```diff
@@ -1,2 +1,146 @@
+This pattern simplifies the integration with any Ethereum based platform. 
+Marmo relayer: (https://github.com/ripio/marmo-relayer). 
+
+# Abstract
+### User pain points:
+> - Users don't want to think about ether
+> - Users want to be able to pay for transactions using whatever they have 
+> - Users don’t want to download apps/extensions (at least on the desktop) to connect to their apps
+
+# Ecosystem Graph
+![](./images/01.png)
+
+###### General WIKI ecosystem.
+- Work in progress.
+###### API layer
+- Marmo relayer doc: (https://github.com/ripio/marmo-relayer/blob/master/README.md)
+###### CORE layer
+- Marmo contracts doc: (https://github.com/ripio/marmo-contracts/blob/master/README.md)
+
+# Features
+- Complete implementation of Intent functionality for Marmo relay.
+- Ethereum wallet support.
+- Comprehensive integration tests demonstrating a number of the above scenarios.
+
+##### Runtime dependencies:
+
+- web3py (https://pypi.org/project/web3/).
+
+# Getting started
+
+##### Prerequisites
+* Python3
+
+Clone the repository and then run:
+
+pip install .
+
+
+### Intent Flowchart
+![](./images/02.png)
+
+###### Dependencies
+> - T0  -> -
+> - T1  -> - 
+> - T2  -> T1
+> - T3  -> T1
+> - T4  -> T1
+> - T5  -> T1, T4
+> - T6  -> T2
+> - T7  -> T2
+> - T8  -> T2
+> - T9  -> T3
+> - T10 -> T4
+> - T11 -> T5
+> - T12 -> T6
+> - T13 -> T7
+> - T14 -> T8
+> - T15 -> T9
+> - T16 -> T10
+> - T17 -> T10
+> - T18 -> T11
+> - T19 -> T13
+> - T20 -> T14
+> - T21 -> T15
+> - T22 -> T18
+> - T23 -> T19
+> - T24 -> T21
+> - T25 -> T22
+> - T26 -> T12, T23, T20, T15
+> - T27 -> T24
+> - T28 -> T25
+> - T29 -> T27
+
+
+### Build a intent
+```python
+from marmopy import Intent,Credentials,IntentAction
+
+class Contract:
+    def __init__(self,contractAddress):
+        self.contractAddress = contractAddress
+        
+    @IntentAction
+    def transfer(self, to:'address', value:'uint256') -> 'bool': pass
+    
+    
+tokenContractAddress = "0x2f45b6fb2f28a73f110400386da31044b2e953d4" #RCN TOKEN
+to = "0x7F5EB5bB5cF88cfcEe9613368636f458800e62CB"
+
+erc20 = Contract(tokenContractAddress)  #example of ERC20 contract with IntentActions injections. 
+
+value = 1
+
+intentAction = erc20.transfer(to,value)
+
+credentials = Credentials("Your private key here");
+
+contractAddress = "0x7c0e959d9b7c82d47c5013832f89282b8b773393"
+
+# Intents ids 32 bits hashestrings dependencies
+dependencies = ["0xee2e1b62b008e27a5a3d66352f87e760ed85e723b6834e622f38b626090f536e",
+                "0x6b67aac6eda8798297b1591da36a215bfbe1fed666c4676faf5a214d54e9e928"]
+
+intent = Intent(intentAction=intentAction, signer=credentials.getAddress(), wallet=contractAddress)
+```
+
+### Sign a intent
+```python
+signedIntent = Intent.signIntent(intent,credentials);
+```
+
+###  Send a intent
+```python
+import requests
+
+requests.post("http://ec2-3-16-37-20.us-east-2.compute.amazonaws.com/relay",json=signedIntent)  #relay url
```

Comment:
The post body has the wrong scheme, it looks something like this:

```json
{
   "to":"0x2f45b6fb2f28a73f110400386da31044b2e953d4",
   "value":0,
   "data":"0xa9059cbb0000000000000000000000007f5eb5bb5cf88cfcee9613368636f458800e62cb0000000000000000000000000000000000000000000000000000000000000000",
   "dependencies":[

   ],
   "signer":"0x0eF72e106b00B46C35Ce47868c44842a2833d7c0",
   "wallet":"0xe392d4ae22ef1bde42cdda75b6e08811ec885134",
   "salt":"0x0000000000000000000000000000000000000000000000000000000000000000",
   "maxGasPrice":9999999999,
   "minGasLimit":0,
   "id":"0x2f15664593bad3be5ae2a3787dfc8eaba9d671ecfd93925310e804f216665523",
   "signature":{
      "r":"0x9f64ee3f46a446a3b1414486758bcf89ec9678294403835e3f37a66864882487",
      "s":"0x1089683359b54d658398f48981f3f3520513eeeda8ce77da66db81cdf907276b",
      "v":"0x1c"
   }
}
```

but it should be like this
```json
{
   "id":"0xacd5d801cecc1790b95c5395e4f48a40d964ae0c6b70051b3c907060e67da079",
   "dependencies":[
      "0xee2e1b62b008e27a5a3d66352f87e760ed85e723b6834e622f38b626090f536e",
      "0x6b67aac6eda8798297b1591da36a215bfbe1fed666c4676faf5a214d54e9e928"
   ],
   "wallet":"0xbbf289d846208c16edc8474705c748aff07732db",
   "tx":{
      "to":"0x2f45b6fb2f28a73f110400386da31044b2e953d4",
      "value":0,
      "data":"0x70a082310000000000000000000000007f5eb5bb5cf88cfcee9613368636f458800e62cb",
      "minGasLimit":300000,
      "maxGasPrice":999999
   },
   "salt":"0x0000000000000000000000000000000000000000000000000000000000000002",
   "signer":"0x9d7713f5048c270d7c1dbe65f44644f4ea47f774",
   "signature":{
      "r":"0xed4eee4e05337e598b1886ef3941f775833f2c7e55bc233fe2cd66d032029649",
      "s":"0x502acc54e8c1ecb4d845a56afb1e43829743c77f3dd60c8b74b25cef33cd1b1e",
      "v":"0x1b"
   }
}
```